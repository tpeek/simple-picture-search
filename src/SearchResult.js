import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import ThumbUp from '@material-ui/icons/ThumbUp';
import Star from '@material-ui/icons/StarRounded';

const useStyles = makeStyles({
  base: {
    position: 'relative',
    maxWidth: 400,
    margin: '40px auto',
    '&::after': {
      content: '" "',
      clear: 'both',
      display: 'table',
    },
  },
  tag: {
    color: 'white',
    backgroundColor: '#00a48f',
    padding: '2px 5px',
    margin: 4,
    display: 'inline-block',
    fontSize: 14,
  },
});

const SearchResult = ({ previewURL, tags, likes, favorites }) => {
  const classes = useStyles();

  return (
    <li className={classes.base}>
      <img src={previewURL} style={{ float: 'left' }} width="200px" />
      <div
        style={{
          float: 'left',
          display: 'flex',
          flexWrap: 'wrap',
          width: 160,
          marginLeft: 16,
        }}
      >
        {tags.split(', ').map((category) => (
          <span className={classes.tag} key={category}>
            {category}
          </span>
        ))}
      </div>
      <div
        style={{
          position: 'absolute',
          bottom: 0,
          left: 224,
          fontSize: 12,
          fontWeight: 'bold',
        }}
      >
        <span>{likes}</span>
        <ThumbUp
          style={{ marginLeft: 5, marginBottom: -2, height: 14, width: 14, marginRight: 32 }}
        />
        <span>{favorites}</span>
        <Star style={{ marginLeft: 5, marginBottom: -3, height: 16, width: 16 }} />
      </div>
    </li>
  );
};

export default SearchResult;
