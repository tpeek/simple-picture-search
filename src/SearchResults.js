import React from 'react';
import SearchResult from './SearchResult';

const SearchResults = ({ results, savedResults, saveResult }) => {
  return (
    <ul>
      {results.map((result) => (
        <SearchResult
          key={result.id}
          saveResult={() => saveResult(result)}
          saved={savedResults.some((existing) => existing.id === result.id)}
          {...result}
        />
      ))}
    </ul>
  );
};

export default SearchResults;
