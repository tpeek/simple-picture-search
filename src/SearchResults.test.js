import React from 'react';
import SearchResults from './SearchResults';
import SearchResult from './SearchResult';

const result = {
  id: 1,
  previewURL: 'https://www.example.com',
  tags: 'animals, tigers, circus',
  likes: 9001,
  favorites: 42,
};

const defaultProps = {
  results: [result],
  savedResults: [result],
  saveResult: jest.fn(),
};

const setup = (props) => shallow(<SearchResults {...defaultProps} {...props} />);

describe('<SearchResults />', () => {
  it('renders snapshot correctly', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('renders an empty list when there are no results', () => {
    const wrapper = setup({ results: [] });

    expect(wrapper).toMatchSnapshot();
  });

  it('sets saved to true when item is in savedResults', () => {
    const wrapper = setup();

    expect(wrapper.find(SearchResult).props().saved).toBe(true);
  });

  it('sets saved to false when item is not in savedResults', () => {
    const wrapper = setup({ savedResults: [] });

    expect(wrapper.find(SearchResult).props().saved).toBe(false);
  });

  it('passes saveResult the result object when called', () => {
    const wrapper = setup();
    wrapper.find(SearchResult).props().saveResult();

    expect(defaultProps.saveResult).toHaveBeenCalledWith(result);
  });
});
