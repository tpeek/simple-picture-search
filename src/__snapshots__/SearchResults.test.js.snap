// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResults /> renders an empty list when there are no results 1`] = `<ul />`;

exports[`<SearchResults /> renders snapshot correctly 1`] = `
<ul>
  <SearchResult
    favorites={42}
    id={1}
    key="1"
    likes={9001}
    previewURL="https://www.example.com"
    saveResult={[Function]}
    saved={true}
    tags="animals, tigers, circus"
  />
</ul>
`;
